tests/functional-swap.rs

```diff
@@ -49,46 +59,158 @@ async fn swap_test_bob_maker() {
 #[tokio::test]
 #[ignore]
 async fn swap_test_alice_maker() {
-    let bitcoin_rpc = bitcoin_setup();
+    let execution_mutex = Arc::new(Mutex::new(0));
+    let bitcoin_rpc = Arc::new(bitcoin_setup());
     let (monero_regtest, monero_wallet) = monero_setup().await;
 
     let (farcasterd_maker, data_dir_maker, farcasterd_taker, data_dir_taker) =
         setup_farcaster_clients().await;
 
-    let (_, bitcoin_address, swap_ids) = make_and_take_offer(
+    let (xmr_dest_wallet_name, bitcoin_address, swap_id) = make_and_take_offer(
         data_dir_maker.clone(),
         data_dir_taker.clone(),
         "Alice".to_string(),
-        &bitcoin_rpc,
-        &monero_wallet,
+        Arc::clone(&bitcoin_rpc),
+        Arc::clone(&monero_wallet),
+        "1 BTC".to_string(),
+        "1 XMR".to_string(),
     )
     .await;
 
     run_swap(
-        swap_ids[0].clone(),
+        swap_id,
         data_dir_maker,
         data_dir_taker,
-        bitcoin_rpc,
+        Arc::clone(&bitcoin_rpc),
         bitcoin_address,
         monero_regtest,
-        monero_wallet,
+        Arc::clone(&monero_wallet),
+        xmr_dest_wallet_name,
+        execution_mutex,
     )
     .await;
 
     cleanup_processes(farcasterd_maker, farcasterd_taker);
 }
 
+#[tokio::test]
+#[ignore]
```

Comment:
Maybe we want to add `#[timeout(600000)]`, a 10 minute timeout (or more). Just in case some tests hang. (`use ntest::timeout;`)

Reply:
18f9662